source/utilities.cc

```diff
@@ -1643,9 +1643,17 @@ namespace aspect
         }
       while (in >> temp_data);
 
+      AssertThrow(in.eof(),
+                  ExcMessage (std::string("While reading the data file '" + filename + "' the ascii data "
```

Comment:
no need for the explicit conversion to `std::string`. The addition of `"..." + std::string` already does that.